paperqa/agents/tools.py

```diff
@@ -1,196 +1,136 @@
+
+        Repeat previous calls with the same query and years to continue a search.
+        This tool can be called concurrently.
+        This tool introduces novel papers, so invoke this tool when just beginning or when unsatisfied with the current evidence.
 
         Args:
-            query: Search keywords followed by optional year or year range
-                (e.g. COVID-19 vaccines, 2022).
+            query: A search query, which can be a specific phrase, complete sentence,
+                or general keywords, e.g. 'machine learning for immunology'. Also can be
+                given search operators.
+            min_year: Filter for minimum publication year, or None for no minimum year.
+                The current year is {current_year}.
+            max_year: Filter for maximum publication year, or None for no maximum year.
+                The current year is {current_year}.
+            state: Current state.
 
         Returns:
             String describing searched papers and the current status.
-        """
+        """  # noqa: E501,W505
+        # Convert to date range (e.g. 2022-2022) if date is present
+        year = (
+            f"{min_year if min_year else ''}-{max_year if max_year else ''}"  # noqa: FURB110
+            if (min_year or max_year)
+            else None
+        )
         # get offset if we've done this search before (continuation of search)
         # or mark this search as new (so offset 0)
-        logger.info(f"Starting paper search for '{query}'.")
-        search_key = query
-        if search_key in self.previous_searches:
+        search_key = query, year
+        try:
             offset = self.previous_searches[search_key]
-        else:
+        except KeyError:
             offset = self.previous_searches[search_key] = 0
 
-        # Preprocess inputs to make ScrapeRequest
-        keywords = query.replace('"', "")  # Remove quotes
-        year: str | None = None
-        last_word = keywords.split(" ")[-1]
-        if re.match(r"\d{4}(-\d{4})?", last_word):
-            keywords = keywords[: -len(last_word)].removesuffix(",").strip()
-            if self.search_type[1]:
-                year = last_word
-                if "-" not in year:
-                    year = year + "-" + year  # Convert to date range (e.g. 2022-2022)
-        index = await get_directory_index(
-            settings=self.shared_state.settings,
-        )
-
+        logger.info(f"{self.TOOL_FN_NAME} starting for query {query!r}.")
+        index = await get_directory_index(settings=self.settings)
         results = await index.query(
-            keywords,
-            top_n=self.shared_state.settings.agent.search_count,
+            query,
+            top_n=self.settings.agent.search_count,
             offset=offset,
             field_subset=[f for f in index.fields if f != "year"],
         )
+        logger.info(
+            f"{self.TOOL_FN_NAME} for query {query!r} returned {len(results)} papers."
```

Comment:
Let's nix tool name in favor of the prior format, it's formatted strangely in the logs.

Reply:
Okay, done